doc/javaapi.md

```diff
@@ -0,0 +1,41 @@
+##SJS Java Api
+
+The Java API in the Spark Job Server is designed to be a more Java friendly API for users to run Spark Jobs
+inside of SJS.
+
+There are a few differences between this API and the standard one.
+
+- The API is entirely Java based
+- Type safety is done via generics instead of type aliases
+- Requires it's own set of Context factories specific to Java (very important)
+
+###Basic Example
+
+```java
+public class JavaHelloWorldJob implements JSparkJob<String> {
```

Comment:
Yes, thanks again for adding this, does the main README have a link to this one?